src/app/core/basic-datatypes/entity/entity.datatype.ts

```diff
@@ -46,19 +47,59 @@ export class EntityDatatype extends StringDatatype {
     super();
   }
 
-  override importMapFunction(
+  /**
+   * Maps a value from an import to an actual entity in the database by comparing the value with the given field of entities.
+   * Handles type conversion between numbers and strings to improve matching.
+   *
+   * @param val The value from an import that should be mapped to an entity reference.
+   * @param schemaField The config defining details of the field that will hold the entity reference after mapping.
+   * @param additional The field of the referenced entity that should be compared with the val.
+   * @returns Promise resolving to the ID of the matched entity or undefined if no match is found.
+   */
+  override async importMapFunction(
     val: any,
     schemaField: EntitySchemaField,
-    additional?: any,
-  ) {
-    if (!additional) {
-      return Promise.resolve(undefined);
+    additional?: string,
+  ): Promise<string | undefined> {
+    if (!additional || val == null) {
+      return undefined;
+    }
+
+    const normalizedVal = this.normalizeValue(val);
+
+    try {
+      const entities = await this.entityMapper.loadType(schemaField.additional);
+      const matchedEntity = entities.find((entity) => {
+        const entityFieldValue = this.normalizeValue(entity[additional]);
+        return entityFieldValue === normalizedVal;
+      });
```

Comment:
```suggestion
    try {
      const entities = await this.entityMapper.loadType(schemaField.additional);
      const matchedEntity = entities.find((entity) => this.normalizeValue(entity[additional]) === this.normalizeValue(val));
```

this is a little shorter and easier to read, maybe.

(probably this change to be done in the IDE to fix lint / code formatting)